File: cs/Doppelganger/SafeCalculator.cs
namespace Doppelganger;

public class SafeCalculator
{
    private readonly IAuthorizer _authorizer;

    public SafeCalculator(IAuthorizer authorizer)
    {
        _authorizer = authorizer;
    }

    public int Add(int x, int y)
    {
        bool authorized = _authorizer.Authorize();
        // BUG! Should be if(!authorized)
        if (authorized)
        {
            throw new Exception("Not authorized");
        }
        return x + y;
    }
}

File: php/src/SafeCalculator.php
<?php

namespace Doppelganger;

use Exception;

final class SafeCalculator
{
    public function __construct(
        private readonly AuthorizerInterface $authorizer
    ) {
    }

    public function add(int $x, int $y): int
    {
        $authorized = $this->authorizer->authorize();

        // BUG! Should be if (!$authorized)
        if ($authorized) {
            throw new Exception('Not authorized');
        }

        return $x + $y;
    }
}
